Mathlib/Algebra/Category/CommAlg/Basic.lean

```diff
@@ -0,0 +1,309 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Christian Merten, Michał Mrugała, Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Christian Merten, Michał Mrugała, Andrew Yang
+-/
+import Mathlib.Algebra.Category.ModuleCat.Basic
+import Mathlib.Algebra.Category.Ring.Under.Basic
+import Mathlib.CategoryTheory.ChosenFiniteProducts
+
+/-!
+# Category of commutative algebras over a commutative ring
+
+We introduce the bundled category `CommAlg` of algebras over a fixed commutative ring `R` along
+with the forgetful functors to `RingCat` and `ModuleCat`. We furthermore show that the functor
+associating to a type the free `R`-algebra on that type is left adjoint to the forgetful functor.
+-/
+
+open CategoryTheory Limits
+
+universe v u
+
+variable {R : Type u} [CommRing R]
+
+variable (R) in
+/-- The category of R-algebras and their morphisms. -/
+structure CommAlg where
+  private mk ::
+  /-- The underlying type. -/
+  carrier : Type v
+  [isRing : CommRing carrier]
+  [isAlgebra : Algebra R carrier]
+
+attribute [instance] CommAlg.isRing CommAlg.isAlgebra
+
+initialize_simps_projections CommAlg (-isRing, -isAlgebra)
+
+namespace CommAlg
+variable {A B C : CommAlg.{v} R}
+
+instance : CoeSort (CommAlg R) (Type v) := ⟨CommAlg.carrier⟩
+
+attribute [coe] CommAlg.carrier
+
+variable (R) in
+/-- The object in the category of R-algebras associated to a type equipped with the appropriate
+typeclasses. This is the preferred way to construct a term of `CommAlg R`. -/
+abbrev of (X : Type v) [CommRing X] [Algebra R X] : CommAlg.{v} R := ⟨X⟩
```

Comment:
Is there a reason why this is preferred instead of `mk`? The latter has the benefit of supporting the anonymous constructor syntax.

Reply:
I believe the issue is that `mk` has `R` implicit. Do we care?

Reply:
I think this pattern is used elsewhere, indeed for explicit arguments.

Reply:
I'm not convinced that the explicit argument is worth loosing the anonymous constructor syntax.

Reply:
If you feel that way then I think you should start by deleting the existing `of`s in mathlib; I think we should aim to be locally consistent first, and globally optimal second.